docs/resources/backup_config.md

```diff
@@ -0,0 +1,117 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "cockroach_backup_config Resource - terraform-provider-cockroach"
+subcategory: ""
+description: |-
+  The backup configuration for a cluster.
+  Each cluster has a backup configuration that determines if backups are enabled,
+  how frequently they are taken and how often they are retained for.  This
+  resource allows configuring those settings.  It's important to note that the
+  existence or lack of this resource doesn't not indicate whether a backup
+  configuration exists or not. Rather, if Terraform is managing it or not.
+  As a result of this, removal of the resource from the Terraform configuration
+  will leave the backup configuration in the state it was before the removal.
+  retention_days Edit Restrictions
+  retention_days can only be set once.  Further changes require opening a support
+  ticket.  For this reason, managing the value via terraform can require special
+  care. Here are some useful tips:
+  * Optionally, you can refrain from including retention_days in the
+  terraform resource to fully rely on server side management of the value and not need any
+  special care.
+  * If the initial value for retention_days in the backupconfig resource is the
+  default value (30), the single edit will not be used up and it can be changed one
+  more time.
+  * If the initial value set for retention_days in the backupconfig resource is
+  not the default, the single edit will be used up on the first apply. Further
+  edits will require a support ticket.
+  Changing the value of retention_days after using your one change will be a
+  multi-step operation. Here are two workflows that will work.  Both of these
+  paths assume you already have a backup config resource that doesn't have any
+  updates left:
+  Change it and then open a ticket before applying:
+  
+  Update retention_days to the new desired value in terraform.Before applying the run, contact support to change the retention_days to the desired value.Apply the changes in terraform, a terraform READ operation will be done and see the existing value and update the tf state.Temporarily remove management of it from tf, update it via ticket and then add it back.
+  
+  Remove management of retention_days from the terraform resourceRun the apply, nothing will change but terraform is no longer managing that value.Open a support ticket to update retention_days to the desired value.Optionally, add retention_days back to the terraform config with the new value and apply the no-op update.
+---
+
+# cockroach_backup_config (Resource)
+
+The backup configuration for a cluster.
```

Comment:
```suggestion
The backup settings for a cluster.
```

In the docs, we are using "backup settings". I don't know if we need to keep this consistent, so this is a total nit.